frontend/components/MainPageV1/Home/AgentInto/AgentRunButton/AgentNotRunningButton.tsx

```diff
@@ -0,0 +1,361 @@
+import { SERVICE_TEMPLATES } from '@/constants/serviceTemplates';
+import { Pages } from '@/enums/Pages';
+import { MasterEoa, MasterSafe } from '@/enums/Wallet';
+import { useBalanceAndRefillRequirementsContext } from '@/hooks/useBalanceAndRefillRequirementsContext';
+import { useBalanceContext } from '@/hooks/useBalanceContext';
+import { useElectronApi } from '@/hooks/useElectronApi';
+import { MultisigOwners, useMultisigs } from '@/hooks/useMultisig';
+import { useNeedsFunds } from '@/hooks/useNeedsFunds';
+import { usePageState } from '@/hooks/usePageState';
+import { useService } from '@/hooks/useService';
+import { useServices } from '@/hooks/useServices';
+import { useSharedContext } from '@/hooks/useSharedContext';
+import {
+  useActiveStakingContractDetails,
+  useStakingContractContext,
+} from '@/hooks/useStakingContractDetails';
+import { useStakingProgram } from '@/hooks/useStakingProgram';
+import { useMasterWalletContext } from '@/hooks/useWallet';
+import { ServicesService } from '@/service/Services';
+import { WalletService } from '@/service/Wallet';
+import { AgentConfig } from '@/types/Agent';
+import { delayInSeconds } from '@/utils/delay';
+import { updateServiceIfNeeded } from '@/utils/service';
+
+import { AgentBusyButton } from './AgentBusyButton';
+
+/**
+ * hook to handle service deployment and starting the service
+ */
+const useServiceDeployment = () => {
+  const { showNotification } = useElectronApi();
+  const { isAgentsFunFieldUpdateRequired } = useSharedContext();
+
+  const { goto: gotoPage } = usePageState();
+  const { masterWallets, masterSafes, masterEoa } = useMasterWalletContext();
+  const {
+    selectedService,
+    setPaused: setIsServicePollingPaused,
+    refetch: updateServicesState,
+    isLoading: isServicesLoading,
+    selectedAgentConfig,
+    selectedAgentType,
+    overrideSelectedServiceStatus,
+  } = useServices();
+  const serviceId = selectedService?.service_config_id;
+
+  const { canStartAgent, isBalancesAndFundingRequirementsLoading } =
+    useBalanceAndRefillRequirementsContext();
+  const { service, isServiceRunning } = useService(serviceId);
+
+  const { setIsPaused: setIsBalancePollingPaused, updateBalances } =
+    useBalanceContext();
+
+  // Staking contract details
+  const {
+    isAllStakingContractDetailsRecordLoaded,
+    setIsPaused: setIsStakingContractInfoPollingPaused,
+    refetchSelectedStakingContractDetails: refetchActiveStakingContractDetails,
+  } = useStakingContractContext();
+  const { selectedStakingProgramId, selectedStakingProgramMeta } =
+    useStakingProgram();
+  const {
+    isEligibleForStaking,
+    isAgentEvicted,
+    isServiceStaked,
+    hasEnoughServiceSlots,
+  } = useActiveStakingContractDetails();
+
+  const { masterSafesOwners } = useMultisigs(masterSafes);
+
+  const { isInitialFunded, needsInitialFunding } = useNeedsFunds(
+    selectedStakingProgramId,
+  );
+
+  const isLoading = useMemo(() => {
```

Comment:
only this separation was added for this component, the rest was copied from MainPage